### DISCUSSION TITLE
What's the best way to access data inside proxy?

### DISCUSSION THREAD

**Discussion by pansensoyi** [alpinejs/alpine#696] [4. General]
Currently i have this loaded as my x-data:
```
function loadProducts() {
        return {
            productSearchTerm: null,
            items: [],
            filteredResults: [],
            adding: false,
            filterProducts() {
                var products = {!!$products!!}
                Object.filter = function (obj, predicate) {
                    return Object.keys(obj).filter(function (key) {
                        return predicate(obj[key]);
                    }).reduce(function (res, key) {
                        return res[key] = obj[key], res;
                    }, {});
                };
                var searchTerm = this.productSearchTerm;
                var filteredItems = Object.filter(products, function (product) {
                    return product.title === searchTerm || product.id === searchTerm
                });
                this.filteredResults = filteredItems


            }
        }
    }
```
My goal is to filter the object based on the search term inputs. So on the input each keydown im calling filterProducts function. Which then excutes the filter logic. My question is this.filteredResults is wrapped in a proxy. How do i access that data and use that in a x-for template?

**Comment by SimoTod** [comment]
Proxy, when used with array and object literals, are transparent object so they act as the object they contain.
`<template x-for="item in filteredResults">`

**Reply by pansensoyi** [reply]
Thanks @SimoTod hmm i thought so but what about if i do filter function for that object?

**Reply by SimoTod** [reply]
@pansensoyi 
There are a few differences and I made some assumptions on your data but this is an example: https://codepen.io/SimoTod/pen/ExKjNeb
Some considerations:
- I wouldn't change the Object prototype. In my opinion, it's hardly a good idea to change the native types, all your Objects in page would inherit that filter function which is not correct.
- You may want to put your unfiltered list in a variable rather the inside the filtering function so you can potentially use it for other stuff.
- I did not notice that your filteredResults was still an object (I assume you have a good reason for not returning an array), so you'll need Object.keys(filteredResults) in your x-for as well.

**Reply by pansensoyi** [reply]
Hi @SimoTod ,

Thanks for this one very helpful.
